data-processing/data_manipulation/llm_api_service/open_ai_service.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2023 KubeAGI.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from langchain.chains import LLMChain
+from langchain.llms import OpenAI
+from langchain.prompts import PromptTemplate
+
+from common import config
+
+
+def init_service(opt={}):
+    """Initialize the OpenAI service.
+    
+    opt is a dictionary object. It has the following keys:
+    api_key: api key;
+    base_url: base url;
+    model: model;
+    model_type: model type;
+    """
+    api_key = opt.get('api_key', config.open_ai_default_key)
+    base_url = opt.get('base_url', config.open_ai_default_base_url)
+    model = opt.get('model', config.open_ai_default_model)
+    model_type = opt.get('model_type', config.open_ai_default_model_type)
+    
+    llm = OpenAI(
+        openai_api_key=api_key, 
+        base_url=base_url,
+        model=model
+    ) 
+
+    return {
+        'llm': llm,
+        'model_type': model_type
+    } 
+
+
+def generate_qa(llm, opt={}):
```

Comment:
In python ,seems that most projects utlize `**kwargs` to accept flexible key-value arguments.

Reply:
@bjwswang  changed as you said.